lib/ansible/modules/database/postgresql/postgresql_db.py

```diff
@@ -495,6 +495,10 @@ def main():
                 rc, stdout, stderr, cmd = method(module, target, target_opts, db, **kw)
                 if rc != 0:
                     module.fail_json(msg=stderr, stdout=stdout, rc=rc, cmd=cmd)
+
+                elif stderr and ('FATAL' in str(stderr) or 'ERROR' in str(stderr)):
+                    module.fail_json(msg=stderr, stdout=stdout, rc=1, cmd=cmd)
```

Comment:
Normally we have an `stdout` and an `stderr` return value. Personally I would prefer to have the `msg` value indicate that an error occurred, and keep the details in `stderr`. Since it may be multiline output I don't know if putting stderr in msg is a clever thing to do :-/

Reply:
@dagwieers good job! Fixed. PTAL, please.
Looks really good
Dump done example:
```
    "msg": "Dump of acme has been done", 
    "rc": 0, 
    "stderr": "pg_dump: server version: 11.1; pg_dump version: 9.2.24\npg_dump: aborting because of server version mismatch\n", 
    "stderr_lines": [
        "pg_dump: server version: 11.1; pg_dump version: 9.2.24", 
        "pg_dump: aborting because of server version mismatch"
    ], 
    "stdout": "", 
    "stdout_lines": []
```
Dump failed example:
```
    "msg": "Dump of database acme failed", 
    "rc": 1, 
    "stderr": "pg_dump: [archiver (db)] connection to database \"acme\" failed: FATAL:  pg_hba.conf rejects connection for host \"[local]\", user \"postgres\", database \"acme\", SSL off\n", 
    "stderr_lines": [
        "pg_dump: [archiver (db)] connection to database \"acme\" failed: FATAL:  pg_hba.conf rejects connection for host \"[local]\", user \"postgres\", database \"acme\", SSL off"
    ], 
    "stdout": "", 
    "stdout_lines": []
```

Reply:
Hmmm, the "Dump of acme has been done" has not actually been done.

>        "pg_dump: aborting because of server version mismatch"

Reply:
Oh;) , sorry, it was early morning... why it returnes 0... Ok

Reply:
Fixed, thank you very much
Dump done:
```
    "msg": "Dump of database acme has been done", 
    "rc": 0, 
    "stderr": "", 
    "stderr_lines": [], 
    "stdout": "", 
    "stdout_lines": []
```
Dump undone:
```
    "msg": "Dump of database acme failed", 
    "rc": 1, 
    "stderr": "pg_dump: [archiver (db)] connection to database \"acme\" failed: FATAL:  pg_hba.conf rejects connection for host \"[local]\", user \"postgres\", database \"acme\", SSL off\n", 
    "stderr_lines": [
        "pg_dump: [archiver (db)] connection to database \"acme\" failed: FATAL:  pg_hba.conf rejects connection for host \"[local]\", user \"postgres\", database \"acme\", SSL off"
    ], 
    "stdout": "", 
    "stdout_lines": []
```